### PR TITLE
[ready] Add introduction part [AK-73]

### PR DESCRIPTION
**Pull request by szarbartosz**

### CODE REVIEW

**Comment by dimamik** [comment]
@szarbartosz 
Hi, can we please move forward with that PR? That would be awesome, thanks a lot 🌞

**Comment by szarbartosz** [comment]
odd behaviour - steps to reproduce:
1. register new user via registration form
2. sign in via login form
3. fill and submit first introduction step (user details - should proceed to it after login - http://localhost:4200/introduction/user-details endpoint)
4. visit http://localhost:4200/login address once again and try to log in

expected result- login works as usual
actual result:
```UserInputError: Invalid login credentials
    at AuthService.<anonymous> (C:\Users\Lenovo\Desktop\inf\inzynierka\agh-kiwis\dist\apps\server\webpack:\agh-kiwis\apps\server\src\auth\auth.service.ts:49:13)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\Lenovo\Desktop\inf\inzynierka\agh-kiwis\node_modules\tslib\tslib.js:115:62)
```

I guess that userUpdate mutation triggered after first introduction step messes something up, but anyway this snippet of code where auth.service fails:
```
const IsPasswordValid = await bcrypt.compare(
      loginInput.password,
      user.password
    );

    if (!IsPasswordValid) {
      throw new UserInputError('Invalid login credentials');
    }
```

will look into it, but if You have any idea whats wrong with it please let me know

File: libs/ui-components/src/lib/Pickers/ColorPicker.tsx [hunk 1]
```diff
@@ -69,7 +76,7 @@ export const ColorPicker: React.FC<ColorPickerProps> = ({
           <ModalHeader>{modalTitle}</ModalHeader>
           <ModalCloseButton />
           <ModalBody>
-            <VStack justify="center">{renderTiles(data.getColors)}</VStack>
+            <VStack justify="center">{renderTiles(data.getCategories)}</VStack>
```
Comment (dimamik) [inline libs/ui-components/src/lib/Pickers/ColorPicker.tsx:79]: Instead let's put `data?.getCategories`

File: libs/ui-components/src/lib/Pickers/ColorPicker.tsx [hunk 2]
```diff
@@ -27,33 +27,40 @@ export const ColorPicker: React.FC<ColorPickerProps> = ({
...
-  const { data, loading } = useGetColorsQuery();
 
-  const renderTiles = (colors: Color[]): JSX.Element => (
-    <SimpleGrid columns={2} spacing={4}>
-      {colors.map((color) => (
+  const { data, loading } = useGetCategoriesQuery();
+
+  const renderTiles = (categories: Category[]): JSX.Element => (
+    <SimpleGrid spacing={4}>
+      {categories.map((category) => (
```
Comment (dimamik) [inline libs/ui-components/src/lib/Pickers/ColorPicker.tsx:38]: `categories?.map`

File: apps/client/pages/introduction/user-details.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                touched={!!touched.name}
+              />
+            </Box>
+            <Box my="4">
+              <InputField
+                name="birthDate"
+                label="Birthdate"
+                type="date"
+                touched={!!touched.birthDate}
+              />
```
Comment (dimamik) [inline apps/client/pages/introduction/user-details.tsx:72]: This should validate to not select time in the future

File: apps/client/pages/introduction/user-details.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    });
+    router.push('/introduction/sleep-preferences');
+  };
+
+  if (loading) {
+    return <CustomSpinner />;
+  }
+  return (
+    <Wrapper>
+      <Logo textVisible={false} />
```
Comment (dimamik) [inline apps/client/pages/introduction/user-details.tsx:47]: Logo should link to the main page

**Review by dimamik — COMMENTED** [review]
Throwing a first bunch of review, in progress of resolving setPassword function

File: apps/client/pages/introduction/user-details.tsx (hunk 4 above)
Comment (szarbartosz) [reply apps/client/pages/introduction/user-details.tsx:47]: I don't agree in this case - this page is one of the introduction steps and I guess that we wouldn't like to send user away to the main page if they accidentally click on it.

Comment (dimamik) [reply apps/client/pages/introduction/user-details.tsx:47]: hmm, makes sense, resolving.

**Review by dimamik — APPROVED** [review]
That is AMAZING!!! 🚀 

GREAT JOB ON THAT TASK!!! 🔥

**Pull request merged**